ppl/src/main/java/org/opensearch/sql/ppl/PPLService.java

```diff
@@ -82,17 +86,31 @@ private PhysicalPlan plan(PPLQueryRequest request) {
     // 1.Parse query and convert parse tree (CST) to abstract syntax tree (AST)
     ParseTree cst = parser.parse(request.getRequest());
     UnresolvedPlan ast = cst.accept(
-        new AstBuilder(new AstExpressionBuilder(), request.getRequest()));
-
+        new AstBuilder(new AstExpressionBuilder(), catalogService, request.getRequest()));
     LOG.info("[{}] Incoming request {}", LogUtils.getRequestId(), anonymizer.anonymizeData(ast));
 
+    StorageEngine storageEngine =
+        catalogService.getStorageEngine(getConnector(ast)).orElse(openSearchStorageEngine);
```

Comment:
Is is possible to resolve StorageEngine in analyze stage? https://github.com/opensearch-project/sql/blob/main/core/src/main/java/org/opensearch/sql/analysis/Analyzer.java#L122

Reply:
If you see the creation of Analyzer in the next line, I am passing down the storage engine in the constructor. So, it resolves to correct storage engine. 
if you are talking about something else, do let me know.

Reply:
should this also be done to SQLService if you are getting connector here? or is it possible to make this logic independent of PPL service or SQL service?

Reply:
https://github.com/opensearch-project/sql/blame/main/sql/src/main/java/org/opensearch/sql/sql/config/SQLServiceConfig.java#L40

Analyzer is already taking storage engine as input in sql service. The storageEngine is always fixed with SQL. Since we don't want to support this in SQL right now, I am not thinking to change code there. 

Is it required for S3?

Reply:
Ideally, PPLService should not aware of which storage engine is used. 
I think In Analyzer, we could parse the tree and create StorageEngine.
```
    LogicalPlan logicalPlan =
        new Analyzer(new ExpressionAnalyzer(repository), catalogService).analyze(
            UnresolvedPlanHelper.addSelectAll(ast),
            new AnalysisContext());
```

Reply:
+1. I think this should happen inside analyzer. Suppose we support join between tables from different data sources, the current implementation won't work.

Reply:
Agreed. will try to change the implementation. 
Few concerns from my side.
OpensearchStorageEngine is not managed by CatalogService as there are problems with nodeClient Object currently. 
I don't want to drastically change SQL code and cause regression.